python/lsst/ts/Dome/task_scheduler.py

```diff
@@ -1,30 +1,36 @@
 import asyncio
 import logging
+import time
 
 run_status_loop = True
-logging.basicConfig(level=logging.INFO)
+logging.basicConfig(
+    format="%(asctime)s:%(levelname)s:%(name)s:%(message)s", level=logging.INFO
+)
 log = logging.getLogger("TaskScheduler")
 
 
 async def schedule_task_periodically(period, task):
     """Schedules a task periodically with no drift.
 
+    I rewrote this example a bit to make it work for me:
+
+    https://stackoverflow.com/a/48204092
+
     Parameters
     ----------
     period: int
         The period in (decimal) seconds at which to schedule the function.
     task: coroutine
         The function to be scheduled periodically.
     """
-    loop = asyncio.get_event_loop()
     log.info(f"run_status_loop = {run_status_loop}")
 
     def g_tick():
-        t = loop.time()
+        t = time.time()
         count = 0
         while run_status_loop:
             count += 1
-            yield max(t + count * period - loop.time(), 0)
+            yield max(t + count * period - time.time(), 0)
```

Comment:
I'm uneasy about this class, especially when precise timing is not necessary. Compare using this class to the standard loop approach:
```
async def background_loop(self):
    while True:
        # do stuff
        await asyncio.sleep(...)
```
Personally I find the above easier to read. if you need precise timing you could write a simple functor that computes how long to sleep (a minor variant of this class), and I *think* the result would still be simpler.

Reply:
I was looking for a way to run a loop in the background and that executes a task on a fixed rate. I agree that there is no need for a very precise timing so I replaced most of the code with the asyncio.sleep call and that's it.